packages/blocks/src/store/selectors.js

```diff
@@ -44,7 +44,7 @@ export const getBlockTypes = createSelector(
 		return Object.values( state.blockTypes ).map( ( blockType ) => {
 			return {
 				...blockType,
-				variations: getBlockVariations( state, blockType.name ),
+				variations: getBlockVariations( state, blockType?.name ),
```

Comment:
We can drop this one as well, `state.blockTypes` is an array

Reply:
How did you test it and come to know that we can drop this one. Indeed, there was no error here! But I added it myself. How do you come to know it is not required?

Reply:
Sorry `state.blockTypes` should be an object rather than an array. We can verify this by looking at the test cases in the block store.

https://github.com/WordPress/gutenberg/blob/7ae4f3d801e65cbc4a936ae0f45058d4392f6ba5/packages/blocks/src/store/test/reducer.js#L26-L30

If we were interested (and needed the change) we can also write our own unit test in `test/selectors.js`

```jsx
	describe( 'getBlockTypes', () => {
		it( 'returns empty array when no types are registered', () => {
			expect( getBlockTypes( { blockTypes: {} } ) ).toEqual( [] );
		} );
	} );
```

And we'd be able to run these locally via 
```
npm run test-unit packages/blocks/src/store/test/selectors
```

See also: https://developer.wordpress.org/block-editor/contributors/code/testing-overview/#javascript-testing